lib/testLauncher/repl.js

```diff
@@ -0,0 +1,174 @@
+const net = require('net');
+const path = require('path');
+const repl = require('repl');
+const colors = require('colors');
+const chokidar = require('chokidar');
+
+const {getShortFunctionBody} = require('../utils/stringUtils');
+const SuitestError = require('../utils/SuitestError');
+const {REPL_START, REPL_STOP, REPL_INIT} = require('../constants/repl');
+const texts = require('../texts');
+
+const exceptionListeners = {};
+const errorEvents = ['uncaughtException', 'unhandledRejection'];
+
+function pauseErrorListeners() {
+	// return;
+	errorEvents.forEach(event => {
+		if (!exceptionListeners[event])
+			exceptionListeners[event] = process.rawListeners(event).map(one => one);
+
+		process.removeAllListeners(event);
+		process.on(event, err => console.error(err));
+	});
+}
+
+function restoreErrorListeners() {
+	errorEvents.forEach(event => {
+		if (exceptionListeners[event]) {
+			process.removeAllListeners(event);
+			exceptionListeners[event].forEach(func => process.on(event, func));
+		}
+	});
+}
+
+/**
+ * Establishes communication channel with the main process and
+ * starts the REPL
+ * @param {Object} opts - repl opts see below
+ * @returns {Promise<any>} - resolves when repl session ends
+ */
+function startRepl(opts) {
+	const {
+		replIpcErrorInChildProcess,
+		replIpcNotAvailable,
+		replIpcNotImplemented,
+	} = texts;
+
+	if (!process.env.REPL_IPC_PORT) {
+		throw new SuitestError(
+			replIpcNotAvailable(),
+			SuitestError.IPC_ERROR
+		);
+	}
+
+	const ipcSocket = net.connect(process.env.REPL_IPC_PORT, 'localhost').unref();
+
+	ipcSocket.on('connect', () => {
+		ipcSocket.write(REPL_START);
+	});
+
+	ipcSocket.on('error', err => {
+		throw new SuitestError(
+			replIpcErrorInChildProcess(err.message || err.code),
+			SuitestError.IPC_ERROR,
+			err
+		);
+	});
+
+	return new Promise(resolve => {
+		ipcSocket.on('data', chunk => {
+			const msg = chunk.toString().split('|');
+			const [ttyFlag, ttyVal] = msg;
+
+			if (ttyFlag === REPL_INIT)
+				return resolve(setupRepl(opts, !!+ttyVal, ipcSocket));
```

Comment:
It resolves with another Promsie, but ipcSocket will keep listening for 'data' ?

Reply:
Not sure what you mean with "another promise" but yes, the socket will remain alive